src/hotspot/share/oops/fieldInfo.hpp

```diff
@@ -220,29 +220,38 @@ class Mapper {
   void map_field_info(const FieldInfo& fi);
 };
 
+#define JUMP_TABLE_STRIDE 16
```

Comment:
How was the threshold of 16 determined?

Reply:
I haven't done any benchmarks looking for the optimal value; this should balance the extra memory footprint vs. improved performance. Also I was hoping to not affect the bulk of Java code; rather optimize "big" classes that show degraded performance due to O(N) lookup.
How exactly could the optimization function look like if we're to weigh in both memory consumption and execution speed?

Reply:
Since this is also referenced by SA, it should be exported in vmStructs and not also defined separately in SA. I believe that you can use something like ObjectMonitor::ANONYMOUS_OWNER as an example of how to do this:

In hotspot:
  static const int64_t ANONYMOUS_OWNER = 1;

In vmstructs:
  declare_constant(ObjectMonitor::ANONYMOUS_OWNER)                        \

In SA:
    ANONYMOUS_OWNER = db.lookupLongConstant("ObjectMonitor::ANONYMOUS_OWNER").longValue();